### ISSUE TITLE
HasFieldsWithSameValues throws NullReferenceExceptio for some null properties of type interface

### ISSUE THREAD

**Issue by gregoiremercier** [tpierrain/NFluent#215]
Hello,

After upgrading from NFluent 2.0 to 2.1, I have the following issue: if a property of type interface is null, NFluent still try to check the sub-properties.

For example:

```
    public class MyClass
    {
        public ISubClass SubClass { get; set; }
    }

    public interface ISubClass
    {
        int MyProperty { get; set; }
    }

    public class SubClass : ISubClass
    {
        public int MyProperty { get; set; }
    }
```

The following test fails
```
        public void Test()
        {
            var first = new MyClass();
            var second = new MyClass();

            Check.That(first).HasFieldsWithSameValues(second);
        }
```

For both instance, the property is null but NFluent still try to access the properties of ISubClass

However if i don't use the interface but directly the class for the property's type, the test is successful

Regards,

**Comment by dupdob** [comment]
Hello
sorry for the inconvenience and thanks for bringing that up. I look into this right now

**Comment by dupdob** [comment]
I confirm I reproduce the issue with the latest version. I have identified and fixed the issue.
A beta will be published in a few minutes

**Comment by dupdob** [comment]
beta is available on the beta feed (through myget)
you have all details here:
https://www.myget.org/feed/dupdobnightly/package/nuget/NFluent
Alternatively, you can also download the package from AppVeyor: https://ci.appveyor.com/project/tpierrain/nfluent/branch/master/artifacts

**Comment by gregoiremercier** [comment]
Downloaded and tested : it works.

Thanks for the quick fix

**Comment by dupdob** [comment]
Thanks for confirming the fix. An official version will be released soon.

**Comment by dupdob** [comment]
Fix released in V 2.1.1
